File: app/src/main/java/forpdateam/ru/forpda/ui/views/FunnyContent.java
package forpdateam.ru.forpda.ui.views;

import android.content.Context;
import android.support.annotation.DrawableRes;
import android.support.annotation.StringRes;
import android.widget.ImageView;
import android.widget.RelativeLayout;
import android.widget.TextView;

import forpdateam.ru.forpda.App;
import forpdateam.ru.forpda.R;

/**
 * Created by radiationx on 06.10.17.
 */

public class FunnyContent extends RelativeLayout {
    private ImageView image;
    private TextView title, desc;

    public FunnyContent(Context context) {
        super(context);
        inflate(context, R.layout.funny_content, this);
        image = (ImageView) findViewById(R.id.funny_image);
        title = (TextView) findViewById(R.id.funny_title);
        desc = (TextView) findViewById(R.id.funny_desc);
    }

    public FunnyContent setImage(@DrawableRes int resId) {
        image.setImageDrawable(App.getVecDrawable(getContext(), resId));
        return this;
    }

    /*public FunnyContent setTitle(String text) {
        title.setText(text);
        title.setVisibility(VISIBLE);
        return this;
    }*/

    /*public FunnyContent setDesc(String text) {
        desc.setText(text);
        desc.setVisibility(VISIBLE);
        return this;
    }*/

    public FunnyContent setTitle(@StringRes int resId) {
        title.setText(resId);
        title.setVisibility(VISIBLE);
        return this;
    }

    public FunnyContent setDesc(@StringRes int resId) {
        desc.setText(resId);
        desc.setVisibility(VISIBLE);
        return this;
    }
}

File: app/src/main/java/forpdateam/ru/forpda/ui/views/FunnyContent.kt
package forpdateam.ru.forpda.ui.views

import android.content.Context
import android.widget.RelativeLayout
import androidx.annotation.DrawableRes
import androidx.annotation.StringRes
import by.kirich1409.viewbindingdelegate.viewBinding
import forpdateam.ru.forpda.databinding.FunnyContentBinding

/**
 * Created by radiationx on 06.10.17.
 */
class FunnyContent(context: Context?) : RelativeLayout(context) {

    private val binding by viewBinding<FunnyContentBinding>(attachToRoot = true)

    fun setImage(@DrawableRes drawableRes: Int): FunnyContent {
        binding.funnyImage.setImageResource(drawableRes)
        return this
    }

    fun setTitle(@StringRes stringRes: Int): FunnyContent {
        binding.funnyTitle.setText(stringRes)
        binding.funnyTitle.visibility = VISIBLE
        return this
    }

    fun setDesc(@StringRes stringRes: Int): FunnyContent {
        binding.funnyDesc.setText(stringRes)
        binding.funnyDesc.visibility = VISIBLE
        return this
    }
}
